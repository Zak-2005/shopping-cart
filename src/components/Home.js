import React from "react";
import { HashRouter, Link } from "react-router-dom";
const Home = () => {
  return (
    <div className="home">
      <h1 className="homeText">
        Your dining experience is about to change forever
      </h1>
      <div>
        <Link to="/shop/products">
          <button type="button" id="shopNow">Shop Now</button>
        </Link>
   
      </div>
    </div>
  );
};
export default Home;
