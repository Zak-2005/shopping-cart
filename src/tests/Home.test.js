import React from "react";
import { render } from "@testing-library/react";
import Home from "../components/Home";
import "@testing-library/jest-dom";
import { HashRouter } from "react-router-dom";

describe("Home Component", () => {
  it("Renders Correct Heading and Button Text", () => {
    
    const { container } = render(<HashRouter><Home /></HashRouter>);
    expect(container).toMatchSnapshot();
    
  });
});
