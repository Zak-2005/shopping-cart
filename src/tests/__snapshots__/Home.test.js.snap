// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Renders Correct Heading and Button Text 1`] = `
<div>
  <div
    class="home"
  >
    <h1
      class="homeText"
    >
      Your dining experience is about to change forever
    </h1>
    <div>
      <a
        href="#/shop/products"
      >
        <button
          id="shopNow"
          type="button"
        >
          Shop Now
        </button>
      </a>
    </div>
  </div>
</div>
`;
